New Editing Section Navigation and Docs (#244)
CloudCannon/platform-documentation: articles/introduction-to-editing.mdx

In the Editing section of our documentation, we cover how to:

* [Create a global configuration file](/documentation/articles/create-a-global-configuration-file/) — create the file at the heart of CloudCannon customization.
* [Define your collections](/documentation/articles/define-your-collections/) — groups of related files that form the start of any new site.
* Editing interfaces — the [Visual](/documentation/articles/introducing-the-visual-editor/), [Content](/documentation/articles/introducing-the-content-editor/), [Data](/documentation/articles/introducing-the-data-editor/), and [Source](/documentation/articles/introducing-the-source-editor/) Editors.
* [Edit your files](/documentation/articles/edit-your-files/) — cooperate on file changes with live editing.
* [Save your changes](/documentation/articles/save-your-changes/) — learn how to review changes made by your team and save back to your git repository.
* [Configure your inputs](/documentation/articles/configure-your-inputs/) — configure your structured data to create an intuitive interface.
* [Create a structure](/documentation/articles/create-a-structure/) — create predefined objects to add to your content.

> * Create a Snippet — create structured components in Markup using [Docusaurus](/documentation/articles/snippets-using-docusaurus-components/), [Eleventy](/documentation/articles/snippets-using-eleventy-shortcodes/), [Hugo Shortcodes](/documentation/articles/snippets-using-hugo-shortcodes/), [MDX](/documentation/articles/snippets-using-mdx-components/), or [Python Markdown](/documentation/articles/snippets-using-python-markdown/).

Reviewer: We should either name these after the SSG (`Eleventy`) or concept within the SSG (`Hugo Shortcodes`)

```suggestion
* Create a Snippet — create structured components in Markup using:
  * [Docusaurus Components](/documentation/articles/snippets-using-docusaurus-components/)
  * [Eleventy Shortcodes](/documentation/articles/snippets-using-eleventy-shortcodes/)
  * [Hugo Shortcodes](/documentation/articles/snippets-using-hugo-shortcodes/)
  * [MDX Components](/documentation/articles/snippets-using-mdx-components/)
  * [Python Markdown](/documentation/articles/snippets-using-python-markdown/).
```

Let’s briefly introduce a few of these topics.

## The Global Configuration File

Unlike a traditional CMS, you can configure CloudCannnon to suit your build tools and code. To maintain this flexibility, CloudCannon does not make assumptions about how you have configured your static site. CloudCannon will need you to clarify a few things about your SSG setup:

* Where are your content files stored? How are they grouped?
* Should CloudCannon allow the creation of new files? If so, how should they be created?
* Do you want your team to be able to edit those files? If so, what options should they have?

This setup information is stored in the CloudCannon global configuration file, saved in the root directory of your repository. Your configuration file can be one of several file types (i.e., `/cloudcannon.config.*`). When CloudCannon searches the root directory for your config file, it will use the first matching file from this list.

1. `cloudcannon.config.json`
2. `cloudcannon.config.yaml`
3. `cloudcannon.config.yml`
4. `cloudcannon.config.js`

> 5. `cloudcannon.config.cjs`

Reviewer: The `.js` and `.cjs` files are not supported in Jekyll. Not sure if that is worth a specific callout

For a more in-depth explanation of Collections, including how to display Collections in the site navigation, control who can create new posts, and customize the default content, name, and editing interface of new files, read our documentation:

* [Define your collections](/documentation/articles/define-your-collections/)

## Supported file types

CloudCannon supports the following file formats for standalone data files or Collections:

* Structured data files: JSON, YAML, TOML, CSV, TSV
* Markup files: HTML, Markdown, MDX
* Combination files: HTML with front matter, Markdown with front matter, MDX with front matter

Structured data files are a great way to store data. Each key can contain strings, numbers, rich text, or even arrays/objects, which can recurse indefinitely. We recommend structured data files for page-building components, navigation, header and footer, site metadata, or any file that does not contain long-form prose.

> Markup files should store more complex data, such as files with multiple lines of text or complex formatting. We recommend markup files for blog posts, articles, or documentation.

Reviewer: I don't know if "more complex data" is the right phrasing here.

I would say something like:

```suggestion
Markup files should store loosely structured content, such as files with multiple lines of text or complex formatting. We recommend markup files for blog posts, articles, or documentation.
```

```suggestion
Markup files should store longer written content, such as files with multiple lines of text or complex formatting. We recommend markup files for blog posts, articles, or documentation.
```

```suggestion
Markup files should store free-form content, such as files with multiple lines of text or complex formatting. We recommend markup files for blog posts, articles, or documentation.
```

> Combination files use structured data and other data in the same file. CloudCannon expects structured data to be at the top of the file in the front matter section. Front matter is separated from other content by a `---` notation or similar. You can edit both parts of a combination file using the Visual or Content Editor. We recommend combination files for pages where you want to attach metadata to your content.

Reviewer: ```suggestion
Combination files use structured data and markup in the same file. CloudCannon expects structured data to be at the top of the file in the front matter section. Front matter is separated from other content by a `---` notation or similar. You can edit both parts of a combination file using the Visual or Content Editor. We recommend combination files for pages where you want to attach metadata to your content.
```

The Visual Editor is an interactive preview of your site. Your team can navigate each page by clicking links as you would on the live site and edit by clicking directly on the content. Developers can determine which page elements are editable and configure the Visual Editor sidebar to alter a file’s front matter. The Visual Editor removes the guesswork of updating your content by showing you exactly what your live site will look like in a true WYSIWYG experience.

The Content Editor is a rich text editor useful for markup and combination files. This editor provides a clean, distraction-free interface for editing content-heavy files like blog posts. Customize your WYSIWYG toolbar to include the tools your content team needs, add Snippets to create a library of custom content blocks, and use the sidebar to edit front matter content — all in the same editor.

The Data Editor is a simple interface for managing structured data. Team members can edit [input](/documentation/articles/configure-your-inputs/) values without being overwhelmed by code syntax. CloudCannon uses the Data Editor in the Visual and Content Editors, where it functions as the sidebar to manage front matter content.

> The Source Editor is an in-browser code editor that is great for making minor changes to any of your files. CloudCannon automatically applies syntax highlighting to your files based on the file extension. This editing interface does not require configuration and will work out of the box for any SSG. This editing interface is available to users with [Owner and Developer permission levels](/documentation/articles/managing-your-team-members/).

Reviewer: ```suggestion
The Source Editor is an in-browser code editor that is great for making minor changes to any of your files. CloudCannon automatically applies syntax highlighting to your files based on the file extension. This editing interface does not require configuration and will work out of the box for any SSG. This editing interface is available to users with [Owner, Developer, and Technical Editor permission levels](/documentation/articles/managing-your-team-members/).
```
